### PR TITLE
test: some react tests are flaky [INFENG-663]

### PR DESCRIPTION
**Pull request by JComins000**
## Ticket

[INFENG-663]

## Description

check the jira ticket for all broken tests covered by this change

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[INFENG-663]: https://hpe-aiatscale.atlassian.net/browse/INFENG-663?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/e2e/fixtures/user.fixture.ts [hunk 1]
```diff
@@ -216,7 +216,7 @@ export class UserFixture {
         activate ? 'User has been activated' : 'User has been deactivated',
       );
       await this.userManagementPage.toast.close.pwLocator.click();
-    }).toPass({ timeout: 15000 });
+    }).toPass({ timeout: 35_000 });
```
Comment (JComins000) [inline webui/react/src/e2e/fixtures/user.fixture.ts:219]: there's a lot of steps in here. need bigger timeout

File: webui/react/playwright.config.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export default defineConfig({
   fullyParallel: !!process.env.CI,
 
   /* https://playwright.dev/docs/test-timeouts#global-timeout */
-  globalTimeout: process.env.PWDEBUG ? 0 : 800_000,
+  globalTimeout: process.env.PWDEBUG ? 0 : 900_000,
```
Comment (JComins000) [inline webui/react/playwright.config.ts:26]: had some teardown failing at 800. we have a ticket to revisit this later

File: webui/react/src/e2e/tests/userManagement.spec.ts [hunk 3]
```diff
@@ -13,13 +12,29 @@ import { repeatWithFallback } from 'e2e/utils/polling';
...
-
 test.describe('User Management', () => {
-  test.beforeEach(loginBeforeEach);
+  test.beforeEach(async ({ auth, dev, page }) => {
+      await dev.setServerAddress();
+      await auth.login();
+      const userManagementPage = new UserManagement(page);
+      await userManagementPage.goto();
+
+      // wait for table to be stable and select page 1
```
Comment (JComins000) [inline webui/react/src/e2e/tests/userManagement.spec.ts:22]: this part is new

File: webui/react/src/e2e/tests/userManagement.spec.ts [hunk 4]
```diff
@@ -59,15 +74,12 @@ test.describe('User Management', () => {
...
-        const userManagementPage = new UserManagement(page);
-        await userManagementPage.goto();
+      test('User table shows correct data', async ({ user }) => {
         await user.validateUser(testUser);
       });
 
-      test('Sign in', async ({ page, auth }) => {
+      test('New user acess', async ({ page, auth }) => {
         const userManagementPage = new UserManagement(page);
-        await page.goto('/');
```
Comment (JComins000) [inline webui/react/src/e2e/tests/userManagement.spec.ts:70]: all these `goto` are in setup now

File: webui/react/src/e2e/tests/userManagement.spec.ts [hunk 5]
```diff
@@ -198,8 +213,11 @@ test.describe('User Management', () => {
             ).toHaveLength(10);
           }).toPass({ timeout: 10_000 });
           // go to page 2 to see users
-          await userManagementPage.table.table.pagination.pageButtonLocator(2).click();
-          await expect(userManagementPage.table.table.rows.pwLocator).toHaveCount(1);
+          expect(async () => {
```
Comment (JComins000) [inline webui/react/src/e2e/tests/userManagement.spec.ts:216]: this started flaking on me. it would load page 2 and reset to page 1

**Review by djanicekpach — APPROVED** [review]

**Pull request merged**